pkg/scheduler/framework/plugins/clusteraffinity/state.go

```diff
@@ -6,43 +6,84 @@ Licensed under the MIT license.
 package clusteraffinity
 
 import (
-	fleetv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"k8s.io/apimachinery/pkg/api/resource"
+
+	clusterv1beta1 "go.goms.io/fleet/apis/cluster/v1beta1"
+	placementv1beta1 "go.goms.io/fleet/apis/placement/v1beta1"
+	"go.goms.io/fleet/pkg/scheduler/framework"
 )
 
-type pluginState struct {
-	// requiredAffinityTerms is a list of processed version of required cluster affinity terms.
-	requiredAffinityTerms AffinityTerms
+type observedMinMaxValues struct {
+	min *resource.Quantity
+	max *resource.Quantity
```

Comment:
how do we plan to support property in the future?

Reply:
I am not sure how about we will support sorting with non-sortable-numeric properties (e.g., strings) either. Our API needs to be updated to support this (user directives required); I assume it would be OK if we revisit this when the support materializes?